src/main/webapp/app/upload/upload.service.ts

```diff
@@ -30,4 +36,44 @@ export class UploadService {
                 })
             );
     }
+
+    public parse(data) {
+        const uploadURL = `${SERVER_API_URL}/api/pool/parse`;
+        return this.httpClient.post<Pool[]>(uploadURL, data, {
+            reportProgress: true,
+            observe: 'response'
+        });
+    }
+
+    public save(data) {
```

Comment:
Parse and save methods should switch handing, you send a file to be parsed and JSON for saving.

Reply:
I don't fully understand what do you mean? and what am I supposed to do?

Reply:
fixed